### PR TITLE
Adds a number of new toolmods, weapons, and some new functionality

### PR DESCRIPTION
**Pull request by Jamini**
## About The Pull Request

The Mekhane Electro-Mace. A second high-tier melee weapon on par with the longsword. Spawns from EoTP.
This is a powered weapon that requires a medium cell.

image

The DSS Gemini. This is Darkflare solution's best attempt at emulating a spider rose. It lacks the per-shot punch of the rose, but fires in two-round bursts to compensate! Comes in stun and kill modes, but no charge mode.

image
image

The FS Barrel Diffuser. This mod doubles your energy gun's capacity by halving its damage. Takes a lens slot like the giga lens
image
image

The collapsing tool kit

This tool mod reduces bulk by 1 at the cost of increasing degradation. Perfect for people who want to stuff a lot of tools in a pouch or bag instead of a tool-belt!

image
image
image

Functional tool attachments! These babies make the attached item gain a (low tier) tool quality! They make the tool a bit bulky and take a slot you could use for precision... but if you want to have a junk tool that does it all... you can!

image
image
image

Brace bars and reinforced plating have been slightly buffed. Brace bars now increase your forcemult by 10% (instead of giving a flat 1 damage) Reinforcements now give 2 damage!

image

## Why It's Good For The Game

Lots of new sidegrades to play with. Buffs for less-popular mods and slight buffs to hammers specifically
Lots of new mods to play with as well! Because we can never have enough of them.
image

## Changelog
```changelog
add: Mekhane Electro-Mace. A powerful blunt melee weapon on par with the longsword in power.
add: DSS Gemini. Darkflare Security Solution's answer to the spider rose. This weapon has two barrels that always fire together. Increasing damage per trigger pull at the cost of stopping power. Uses medium cells.
add: Barrel diffuser. This lens augment for energy weapons doubles the length of your battery, but also halves the damage output of your weapon.
add: Folding Kit. This kit allows you fold your weapon down smaller due to cleverly-placed hingest and screws. Reduces bulk by 1 and increases degradation by 100%
add: sharp edge, a new augment that gives a tool 10 cutting and 5 wire cutting at the cost of 1 bulk
add: bolt gripper, a new augment that gives a tool 10 bolt turning at the cost of 1 bulk
add: multi-tool attachment, a new augment that gives a a tool 5 pulsing at the cost of 1 bulk
add: prying head, new augment that gives a tool 10 prying at the cost of 1 bulk
add: screwdriver attachment, a new augment that gives a tool 5 screw driving at the cost of 1 bulk
add: adaz head, a new augment that gives 10 digging, 10 shoveling, and 5 excavation at the cost of 1 bulk
fix: toolmods that add tool qualities now function properly. I.E. The Flat Surface now properly gives hammering 10
balance: Brace bars now give 10% force instead of a flat 1.
balance: Brace bars no longer fit on saws, but do fit on hammers.
balance: Reinforced plating now gives 2 force instead of 1
balance: Flat Surface now adds 1 bulk instead of -10% workspeed
balance: Theoretical paint is now limited between -15 to 15 precision. Down from -20 to 20
balance: Theoretical paint is now limited to between -10 to 10 extra force. Down from -20 to 20

```

### CODE REVIEW

**Comment by bearrrrrrrr** [comment]
POGtanium

File: zzzz_modular_occulus/code/game/objects/items/weapons/tools/mods/mod_types.dm [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	price_tag = 120
+
+/obj/item/weapon/tool_upgrade/augment/folding/New()
+	..()
+	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
+	I.tool_upgrades = list(
+	UPGRADE_DEGRADATION_MULT = 2,
+	UPGRADE_BULK = -1
+	)
+	I.prefix = "Collapsible"
```
Comment (zackerycodestoo) [inline zzzz_modular_occulus/code/game/objects/items/weapons/tools/mods/mod_types.dm:16]: Does this append a capitalized 'Collapsible' to the item name? eg. 'the sanctified edged wrenching Collapsible shovel'

**Review by zackerycodestoo — CHANGES_REQUESTED** [review]
I like the adaz -- what's an adaz, just out of curiosity? I thought maybe 'adze' but that's for woodcutting.

There's only one concern in that grammar might look a bit odd if the folding augment adds a capitalized Collapsible -- see comment

File: zzzz_modular_occulus/code/game/objects/items/weapons/tools/mods/mod_types.dm (hunk 1 above)
Comment (Jamini) [reply zzzz_modular_occulus/code/game/objects/items/weapons/tools/mods/mod_types.dm]: fixed

**Comment by Jamini** [comment]
> I like the adaz -- what's an adaz, just out of curiosity? I thought maybe 'adze' but that's for woodcutting.
> 
> There's only one concern in that grammar might look a bit odd if the folding augment adds a capitalized Collapsible -- see comment

It's a spelling error. You are correct.

Adze are cutting/digging tools, both. Also used for woodcarving.

**Review by Michiyamenotehifunana — APPROVED** [review]
Everything looks about right.

**Pull request merged**